examples/custom-literals.kl

```diff
@@ -0,0 +1,21 @@
+#lang "prelude.kl"
+
+(import (shift "prelude.kl" 1))
+(import "define-syntax-rule.kl")
+
+
+(define-macro (my-integer n)
+  (pure `(+ 1 (#%integer-literal ,n))))
```

Comment:
You sneaky person. I was about to react negatively to the test, and then I saw what was going on :-)

Reply:
It was the simplest possible non-default interpretation of integer literals I could think of. Do you have a less sneaky non-default implementation in mind? We don't have type classes yet, so I couldn't use `(fromInteger (#%integer-literal ,n))`.